node/router/src/inbound.rs

```diff
@@ -340,25 +340,42 @@ pub trait Inbound<N: Network>: Reading + Outbound<N> {
 
         // Truncate and convert to socket addrs.
         peers.truncate(MAX_PEERS_TO_SEND);
-        let peers = peers.into_iter().map(|peer| peer.listener_addr).collect();
+        let peers = peers.into_iter().map(|peer| (peer.listener_addr, peer.last_height_seen)).collect();
 
         // Send a `PeerResponse` message to the peer.
         self.router().send(peer_ip, Message::PeerResponse(PeerResponse { peers }));
         true
     }
 
     /// Handles a `PeerResponse` message.
-    fn peer_response(&self, _peer_ip: SocketAddr, peers: &[SocketAddr]) -> bool {
+    fn peer_response(&self, _peer_ip: SocketAddr, peers: &[(SocketAddr, Option<u32>)]) -> bool {
         // Check if the number of peers received is less than MAX_PEERS_TO_SEND.
         if peers.len() > MAX_PEERS_TO_SEND {
             return false;
         }
-        // Filter out invalid addresses.
+        // Prepare a height filtering closure.
+        let own_height = self.router().ledger.latest_block_height();
+        let height_check = |peer_height: Option<u32>| -> bool {
```

Comment:
Do we use `MINIMUM_NUMBER_OF_PEERS` for anything other than pruning peers? 

If so, this may prevent nodes that are sufficiently caught up (close to tip) from finding peers to connect to.

Reply:
@ljedrz you mentioned the complexity is annoying. But maybe whenever we fetch peers, and also periodically, we can re-order the peer pool by observed height... Or we keep two separate peer_pools: one where we have an observation above our height, and one for the rest.